tests/unit/no-top-level-side-effects.test.ts

```diff
@@ -2462,7 +2484,101 @@ const invalid: RuleTester.InvalidTestCase[] = [
         }
       ]
     }
-  ]
+  ],
+  ...[
+    {
```

Comment:
Please add a test case for:

- An object with a key-value pair where the value is a derived value (e.g. `3 + 14`).
- An object with a key-value pair where the value is an IIFE.
- An object with a key-value pair where the value is an `await`ed call.
- Assigning an object to `module.exports` with a key-value pair where the value is a function call when `options: [options.commonjs]`.
- An object with a dynamic key (e.g. `{[foo]:"bar"}`).

Reply:
will do - I'm kinda busy with work on most week days - I can continue working on this when I'm free, thanks for the review!